src/flowcean/transforms/one_cold.py

```diff
@@ -0,0 +1,153 @@
+import logging
+from collections.abc import Iterable
+from typing import Any, Self, override
+
+import polars as pl
+
+from flowcean.core.transform import Transform
+from flowcean.transforms.one_hot import (
+    NoCategoriesError,
+    NoMatchingCategoryError,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class OneCold(Transform):
+    """Transforms integer features into a set of binary one-cold features.
+
+    Transforms integer features into a set of binary one-cold features. The
+    original integer features are dropped and are not part of the resulting
+    data frame.
+
+    As an example consider the following data
+
+    feature |
+    :------:|
+     0      |
+     1      |
+     2      |
+     1      |
+     5      |
+
+    When the one-cold transformation is applied, the result is as follows
+
+    feature_0 | feature_1 | feature_2 | feature_5
+    ----------|-----------|-----------|----------
+     0        | 1         | 1         | 1
+     1        | 0         | 1         | 1
+     1        | 1         | 0         | 1
+     1        | 0         | 1         | 1
+     1        | 1         | 1         | 0
+    """
+
+    feature_category_mapping: dict[str, dict[str, Any]]
+    check_for_missing_categories: bool
+
+    def __init__(
+        self,
+        feature_categories: dict[str, list[Any]],
+        *,
+        check_for_missing_categories: bool = False,
+    ) -> None:
+        """Initializes the One-Hot transform.
+
+        Args:
+            feature_categories: Dictionary of features and a list of
+                categorical values to encode for each.
```

Comment:
Provide an example for the dictionary? (e.g., with respect to the example above)

Reply:
Added a code example to the "main-docstring".